### PR TITLE
use SSL proxy for remote backends

### PR DESCRIPTION
**Pull request by kylecesmat**
Dev Server doesn't like using secure backends

### CODE REVIEW

**Comment by kylecesmat** [comment]
:pray:

**Comment by kylecesmat** [comment]
:star:

File: dev-server.js [hunk 1]
```diff
@@ -35,8 +37,14 @@ if (!appConfig.proxy.hostname) {
   throw new Error('API proxy hostname not set in configuration');
 }
 
-app.use(appConfig.api.prefix, proxy('http://' + appConfig.proxy.hostname, {
-  limit: '50mb',
+var proxyUri = 'https://' + appConfig.proxy.hostname + ':443';
+
+if (__BACKEND__.match(/\.vm$/)) {
+    proxUri = 'http://' + appConfig.proxy.hostname;
```
Comment (chrisheninger) [inline dev-server.js]: typo in proxy

Comment (baohx2000) [inline dev-server.js]: @kylecesmat :point_up:

File: dev-server.js [hunk 2]
```diff
@@ -35,8 +37,14 @@ if (!appConfig.proxy.hostname) {
   throw new Error('API proxy hostname not set in configuration');
 }
 
-app.use(appConfig.api.prefix, proxy('http://' + appConfig.proxy.hostname, {
-  limit: '50mb',
+var proxyUri = 'https://' + appConfig.proxy.hostname + ':443';
+
+if (__BACKEND__.match(/\.vm$/)) {
```
Comment (spruce-bruce) [inline dev-server.js]: `__BACKEND__` will be undefined in dev builds.

This block was updated on our project:

```
var proxyUri = 'http://' + appConfig.proxy.hostname;
if (__BACKEND__ && !__BACKEND__.match(/\.vm$/)) {
    proxyUri += ':443';
}
```

**Comment by nearengine** [comment]
:+1:

**Pull request merged**